### PR TITLE
Make example panic if advertise fails

### PR DESCRIPTION
**Pull request by lulf**

### CODE REVIEW

File: examples/apps/src/ble_bas_peripheral.rs [hunk 1]
```diff
@@ -74,7 +74,11 @@ where
                     // then return to advertising state.
                     select(connection_task, counter_task).await;
                 }
-                Err(_) => info!("[adv] error"),
+                Err(e) => {
+                    #[cfg(feature = "defmt")]
+                    let e = defmt::Debug2Format(&e);
```
Comment (plaes) [inline examples/apps/src/ble_bas_peripheral.rs:79]: Would this work even without defmt?

Comment (lulf) [reply examples/apps/src/ble_bas_peripheral.rs:79]: I'm not sure what you mean, because it does compile on the non-defmt examples

Comment (plaes) [reply examples/apps/src/ble_bas_peripheral.rs:79]: Ok, so `defmt::Debug2Format` fixes the value for panic formatter.. Apologies for the noise.

**Review by plaes — APPROVED** [review]

**Pull request merged**